Fix "Multiple VM connectivity" tests to allow any mtu > 1000

oshoval opened the pull request:
Fix "Multiple VM connectivity" tests to allow any mtu > 1000 

```release-note
None
```

oshoval commented:
according https://storage.googleapis.com/kubevirt-prow/reports/flakefinder/kubevirt/kubevirt/flakefinder-2020-02-15-168h.html

looks like pull-kubevirt-e2e-kind-k8s-sriov-1.14.2 is skipped ?

oshoval commented:
/test pull-kubevirt-e2e-k8s-1.17.0

oshoval commented:
/test pull-kubevirt-e2e-k8s-1.17.0

oshoval commented:
according one of the runs it looks like those 4 tests that i work on are working

https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/3075/pull-kubevirt-e2e-k8s-1.17.0/1229016525357191168

tests 1539-1542
but i will refactor the code to use mtu > 1000 as an acceptable prerequisites as suggested by Dan

if needed i will split it to 2 PRs since they the 2nd one is  dependent on the 1st one

oshoval commented:
/hold
we need to groom before we merge it
this will be an helper for fixing tests of k8s-1.17.0 as groomed in the sprint

oshoval commented:
/test pull-kubevirt-e2e-k8s-1.17.0

yossisegev commented on tests/vmi_networking_test.go:
```diff
@@ -189,29 +189,19 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
-			expectedMtuString := ""
+			output = strings.TrimSuffix(output, "\n")
+			mtu, err := strconv.Atoi(output)
+			Expect(err).ToNot(HaveOccurred())
```
I'm not sure that an error of strconv.Atoi is something we should trace with gomega's (the test framework) Expect, and hence fail the test. What is the meaning of Atoi returning an error?
I might be wrong here, but it's worth checking.

oshoval replied:
for example i had to use TrimSuffix, because there was \n and then the atoi failed
we need a valid value for mtu, so it can hint good if there isnt imo, what do you think?

yossisegev replied:
OK, makes sense.
So the changes in vmi_networking_test.go LGTM.

oshoval commented:
cleaned, fixed the test , checked with bump, 
removed now the bump since it will be part of Daniel PR

phoracek commented on tests/vmi_networking_test.go:
```diff
@@ -189,29 +189,19 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 				vmiPod,
 				"compute",
-				[]string{"ip", "address", "show", "k6t-eth0"},
+				[]string{"cat", "/sys/class/net/k6t-eth0/mtu"},
```
Why do we need this change? `ip address show` is not working anymore?

oshoval replied:
it does, but we just want the mtu, so no need to parse the whole line
it was Suggested by Dan

oshoval commented:
travis fails @ make apidocs
its just because 

Failed to get resource: GET. [HTTP HTTP/1.1 504 Gateway Time-out: https://jcenter.bintray.com/io/swagger/swagger-models/1.5.12/swagger-models-1.5.12.pom]

how can we rerun it please ?

oshoval commented:
@dhiller 
Travis passed, thanks

dankenigsberg commented:
@oshoval the title of this PR is now out of date, I believe.

oshoval commented:
> @oshoval the title of this PR is now out of date, I believe.

hmm by this title i mean a range ( > 1000) , instead of a scalar list, 
what would you suggest please to write ?
also the description has some more info

dankenigsberg commented:
> > @oshoval the title of this PR is now out of date, I believe.
> 
> hmm by this title i mean a range ( > 1000) , instead of a scalar list,
> what would you suggest please to write ?
> also the description has some more info

ah, got it. I think that 
```
Fix "Multiple VM connectivity" tests to allow any mtu>1000
```
would be shorter and simpler.

phoracek commented:
@oshoval when you fix PR title+description, you should reflect that in the commit title+description.

oshoval commented:
> @oshoval when you fix PR title+description, you should reflect that in the commit title+description.

commit say what went bad and how we fixed it actually will little more words, 
or you prefer it to be changed as well ? 

mtu was 1440 and wasnt part of the supported mtus
added a supported range of mtu instead of a discrete list
(mtu > 1000), because there isnt really a need for a discrete list

oshoval commented:
/hold cancel

phoracek commented:
/lgtm

dhiller commented on tests/vmi_networking_test.go:
```diff
@@ -189,29 +189,19 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 
-			for _, expectedMtu := range possibleMtus {
-				payloadSize = expectedMtu - ipHeaderSize
+			Expect(mtu > 1000).To(BeTrue())
```
nit: for a more fluent reading I'd recommend using

`Expect(mtu > 1000).To(BeNumerically(">", 1000))`

next time :)
